app/data_models/supplementary_data_store.py

```diff
@@ -114,11 +114,21 @@ def get_data(
         For example if you wanted the identifier for the first item in "some_list"
         it would be get_data(identifier="some_list", selectors=["identifier"], list_item_id=list_item_id-1)
         """
-        if self.is_data_repeating(identifier) and not list_item_id:
-            raise InvalidSupplementaryDataSelector(
-                f"Cannot reference items from `{identifier}` outside a repeat"
-            )
-
+        list_items_mapping = self.list_lookup.get(identifier)
+        if (
+            not list_item_id and list_items_mapping
+        ):  # check if we are NOT in repeat, the other method, is_data_repeating(), cannot be used here as it always evaluates to True
+            values = []
+            for list_item_id_key in list_items_mapping:
+                list_item_id = list_items_mapping[list_item_id_key]
+                values.append(self.resolve_value(identifier, selectors, list_item_id))
+            return values
```

Comment:
Few things.

- Use the list look up only when needed, i.e if we don't have list item id
- Use `.values` to avoid looping over the list, to then just fetch the values, that why `.values()` is useful
- This doesn't handle a value for one of the list not existing. We would need to handle that, as we handle it when fetching answers beloning to a list.
- Optional, use a comphrension

```py
        if not list_item_id and (
            list_item_ids := self.list_lookup.get(identifier, {}).values()
        ):
            return [
                value
                for _list_item_id in list_item_ids
                if (value := self.resolve_value(identifier, selectors, _list_item_id))
            ]

        return self.resolve_value(identifier, selectors, list_item_id)
```

You will need to update the test to ensure if the value is missing for one of the item, then it is not rendered. You will need to update the SDS dataset if we don't have an existing example we can use.

Reply:
Re: the above code should check the `value is not None` instead of `if value`, otherwise it will discard falsey value such as `0`.

Reply:
I had to leave list comprehension and make it normal loop, as it was not allowing me to add this logic. If I added `if value := .... and value is not None` it was evaluating it and adding `True` and `False` to the list of values, same with `if value := .... is not None`. So opted for normal for loop instead.